src/Hint/NewType.hs

```diff
@@ -27,6 +27,15 @@ data A = A () -- newtype A = A ()
 newtype Foo = Foo Int deriving (Show, Eq) --
 newtype Foo = Foo { getFoo :: Int } deriving (Show, Eq) --
 newtype Foo = Foo Int deriving stock Show
+data instance Foo Int = Bar Bool -- newtype instance Foo Int = Bar Bool
+data instance Foo Int = Bar {field :: Bool} -- newtype instance Foo Int = Bar {field :: Bool}
+data instance Foo Int = Bar {field :: Int#}
+data instance Foo Int = Bar
+data instance Foo Int = Bar {field1 :: Bool, field2 :: ()}
+newtype instance Foo Int = Bar Bool deriving (Show, Eq) --
```

Comment:
Why the trailing comment? Does that mean anything? Does removing them make it still work? (I note some earlier in the file, can those be deleted too?)

Reply:
I have no clue, I thought this was some special syntax for the tests that meant something like "a hint should trigger, but it doesn't matter what the output of the hint is"

Reply:
From the `README`:

The general syntax is lhs -- rhs with lhs being the expression you expect to be rewritten as rhs. The absence of rhs means you expect no hints to fire. 

.. but we do want hints to fire here, telling the user to use `DerivingStrategies`, right?

Reply:
The commit from which I copied the test structure, if it's any help:
```diff
tree 4c23545e0e5a82f1a5b4a30563a1330ce268aa7a
parent 98298fd75be1d279bc65eac48e87fd6f3e70b1c0
author Neil Mitchell <ndmitchell@gmail.com> Mon Jul 1 21:05:21 2019 +0100
committer Neil Mitchell <ndmitchell@gmail.com> Mon Jul 1 21:05:21 2019 +0100

Fix up the newtype tests


diff --git a/src/Hint/NewType.hs b/src/Hint/NewType.hs
index 64ad3ffd..ea94419d 100644
--- a/src/Hint/NewType.hs
+++ b/src/Hint/NewType.hs
@@ -19,8 +19,8 @@ data A = A {b :: !C} -- newtype A = A {b :: C}
 data A = A Int#
 {-# LANGUAGE UnboxedTuples #-}; data WithAnn x = WithAnn (# Ann, x #)
 data A = A () -- newtype A = A ()
-newtype Foo = Foo Int deriving (Show, Eq) -- newtype Foo = Foo Int deriving newtype (Show, Eq)
-newtype Foo = Foo { getFoo :: Int } deriving (Show, Eq) -- newtype Foo = Foo { getFoo :: Int } deriving newtype (Show, Eq)
+newtype Foo = Foo Int deriving (Show, Eq) --
+newtype Foo = Foo { getFoo :: Int } deriving (Show, Eq) --
 newtype Foo = Foo Int deriving stock Show
 </TEST>
 -}
```

Reply:
That's a pretty useless commit message 😆 - happy enough to let sleeping dogs lie here. I think it may well be saying "there's a hint, i don't care what". Which is a bit poor form, but happy to go with it.

Reply:
(Poor form = from me, originally, not complaining at you for copying me!)